java-frontend/src/main/java/org/sonar/java/se/checks/ConditionAlwaysTrueOrFalseCheck.java

```diff
@@ -76,7 +76,7 @@ public void checkEndOfExecution(CheckerContext context) {
 
   private static Set<List<JavaFileScannerContext.Location>> flowFromNode(ExplodedGraph.Node node, boolean conditionIsAlwaysTrue) {
     List<Class<? extends Constraint>> domains = Lists.newArrayList(ObjectConstraint.class, BooleanConstraint.class);
-    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node.parent(), node.programState.peekValue(), domains);
+    Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(node, node.programState.peekValue(), domains);
```

Comment:
This modification looks a bit suspicious without a testcase and definitely does not sound like a quality flaw fix.